fs/config/configfile/configfile.go

```diff
@@ -107,8 +107,27 @@ func (s *Storage) Save() error {
 		return fmt.Errorf("failed to save config file, path is empty")
 	}
 	fs.Debugf(nil, "Saving config file: %v", configPath)
-	dir, name := filepath.Split(configPath)
-	err := file.MkdirAll(dir, os.ModePerm)
+
+	configPathReal := configPath
+	info, err := os.Lstat(configPath)
```

Comment:
You'll probably be better off using [filepath.EvalSymlinks](https://pkg.go.dev/path/filepath?utm_source=godoc#EvalSymlinks)

Note that the argument it it should be absolute otherwise it won't resolve relative symlinks properly.

I think you can call it even if the argument isn't a symlink which will simplify the code a bit.

Reply:
I started out with EvalSymlinks. Problem was that if the target did not exist, it just returned an error. While with Lstat + Readlink I was able to get the target path even if it does not exist.

Reply:
And also the golang issue 40180 scared me... :) (deliberately not linking to it, I'm find the github cross-linking of all mentions a bit too much sometimes...)

Reply:
Sure, very similar to what I had pushed in an update just before your comment (I assume you hadn't looked at that - or else I'm really confused...). I think this will be a good improvement and as far as I can tell would solve most cases where user's have symlinked the config file, without diving too deep into the rabbit hole with potential for introducing new problems (especially on windows and maybe in combination with network paths, judging from the EvalSymlinks discussions).